From 8133eceab8598cd00da2980cb3760ee2fe869ddb Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Mon, 9 Dec 2024 11:02:21 -0800
Subject: [PATCH] Avoid null pointer dereference in
 ct_font_descriptor_get_traits.

CFDictionaryGetValue can return NULL and CFNumberGetValue doesn't
null-check.
---
 pango/pangocoretext-fontmap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pango/pangocoretext-fontmap.c b/pango/pangocoretext-fontmap.c
index 2eeaf1ae..34ec7bd9 100644
--- a/pango/pangocoretext-fontmap.c
+++ b/pango/pangocoretext-fontmap.c
@@ -267,7 +267,7 @@ ct_font_descriptor_get_traits (CTFontDescriptorRef desc)
    */
   dict = CTFontDescriptorCopyAttribute (desc, kCTFontTraitsAttribute);
   cf_number = (CFNumberRef)CFDictionaryGetValue (dict, kCTFontSymbolicTrait);
-  if (!CFNumberGetValue (cf_number, kCFNumberSInt64Type, &traits))
+  if (!(cf_number && CFNumberGetValue (cf_number, kCFNumberSInt64Type, &traits)))
     traits = 0;
   CFRelease (dict);
 
-- 
2.39.5 (Apple Git-154)

